src/main/kotlin/com/spectralogic/bp/bench/cli/AzSeekableByteChannel.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * ****************************************************************************
+ *   Copyright 2014-2019 Spectra Logic Corporation. All Rights Reserved.
+ * ***************************************************************************
+ */
+
+package com.spectralogic.bp.bench.cli
+
+import java.nio.ByteBuffer
+import java.nio.channels.SeekableByteChannel
+
+class AzSeekableByteChannel : SeekableByteChannel {
+
+    private var size = 0L
+    private var position = 0L
+    private var open = true
+    private var char = 'a'
+
+    override fun read(dst: ByteBuffer): Int {
+        val buffer = dst.asCharBuffer()
+        val amountToRead = Math.min(buffer.remaining(), buffer.limit())
+        val generator = generateSequence {
```

Comment:
It seems like you should create this once for the class instance, rather than on every read call.  This could be a source of a lot of unneeded work.